src/world.js

```diff
@@ -231,6 +232,112 @@ $.extend( $.World.prototype, $.EventSource.prototype, /** @lends OpenSeadragon.W
+        // const promise = this.requestTileInvalidateEvent(priorityTiles, tStamp, restoreTiles);
+        // return promise.then(() => this.requestTileInvalidateEvent(this.viewer.tileCache.getLoadedTilesFor(null), tStamp, restoreTiles));
+
+        // Tile-first retrieval fires computation on tiles that share cache, which are filtered out by processing property
+        return this.requestTileInvalidateEvent(this.viewer.tileCache.getLoadedTilesFor(null), tStamp, restoreTiles);
+
+        // Cache-first update tile retrieval is nicer since there might be many tiles sharing
+        // return this.requestTileInvalidateEvent(new Set(Object.values(this.viewer.tileCache._cachesLoaded)
+        //     .map(c => !c._destroyed && c._tiles[0])), tStamp, restoreTiles);
+    },
+
+    /**
+     * Requests tile data update.
+     * @function OpenSeadragon.Viewer.prototype._updateSequenceButtons
+     * @private
+     * @param {Iterable<OpenSeadragon.Tile>} tilesToProcess tiles to update
+     * @param {Number} tStamp timestamp in milliseconds, if active timestamp of the same value is executing,
+     *   changes are added to the cycle, else they await next iteration
+     * @param {Boolean} [restoreTiles=true] if true, tile processing starts from the tile original data
+     * @param {Boolean} [_allowTileUnloaded=false] internal flag for calling on tiles that come new to the system
+     * @fires OpenSeadragon.Viewer.event:tile-invalidated
+     * @return {OpenSeadragon.Promise<?>}
+     */
+    requestTileInvalidateEvent: function(tilesToProcess, tStamp, restoreTiles = true, _allowTileUnloaded = false) {
+        const tileList = [],
+            markedTiles = [];
+        for (const tile of tilesToProcess) {
+            // We allow re-execution on tiles that are in process but have too low processing timestamp,
+            // which must be solved by ensuring subsequent data calls in the suddenly outdated processing
+            // pipeline take no effect.
+            if (!tile || (!_allowTileUnloaded && !tile.loaded) || tile.transforming) {
+                continue;
+            }
+            const tileCache = tile.getCache(tile.originalCacheKey);
+            if (tileCache.__invStamp && tileCache.__invStamp >= tStamp) {
+                continue;
+            }
+
+            for (let t of tileCache._tiles) {
+                // Mark all related tiles as processing and cache the references to unmark later on,
+                // last processing is set to old processing (null if finished)
+                t.lastProcess = t.processing;
+                t.processing = tStamp;
+                markedTiles.push(t);
+            }
+
+            tileCache.__invStamp = tStamp;
+            tileList.push(tile);
+        }
+
+        if (!tileList.length) {
+            return $.Promise.resolve();
+        }
+
+        // We call the event on the parent viewer window no matter what
+        const eventTarget = this.viewer.viewer || this.viewer;
```

Comment:
I try hard to call everyhithing from the outer context, e.g. that invalidation event is called always on the top level viewer instance, and that viewer.invalidate() is called also on the underlying navigator. However, it seems events like `tile-loaded` do not do this, and expect anyone to also call these on `viewer.navigator` if they want them updated too. More consitent design would be to make stuff separate and do not update viewport when viewer is invalidated. But then it complicates the usage.

Reply:
Yes, but it is also worth considering keeping stuff consitent. If you check `world` implementation, basically anything is viewer reference exclusive. IMHO tile-loaded for example is called only on the viewer instance in question, so you have to bind to `viewer` and `viewer.navigator` to get the event for all tiles in the system? I felt like anyhing you want to perform on navigator must be explicitly allowed. Why not then doing the same here - if you want to invalidate navigator, call it on navigator... etc..

Reply:
True... we might as well continue to keep the navigator separate like that, to stay consistent.